.github/workflows/main.yml

```diff
@@ -26,22 +33,44 @@ jobs:
           cache-dependency-path: |
             **/setup.cfg
             **/requirements*.txt
+            **/test-requirements*.txt
+
       - name: Install dependencies
-        run: |
-          python -m pip install -r test-requirements.txt
-      - name: Test
-        run: python -m pytest
-      - name: Flake8
+        run: pip install -r test-requirements.txt --upgrade pip
+
+      - if: matrix.python-version == '3.10'
```

Comment:
Any reason we are targeting 3.10 for these runs instead of 3.12?

Reply:
Just to ensure they're running in our minimum supported environment in the event the tools might default to adopting language-specific features of newer runtimes. (I'm not aware of that being the case here, but just a potential edge case I've encountered in the PHP world.)